src/main.c

```diff
@@ -602,13 +553,9 @@ init_tty(void)
 		tty_putp(enter_ca_mode);
 
 	/* Emit enough lines to fit all choices. */
-	SLIST_FOREACH(choice, choices, choices) {
-		choices_count++;
+	for (i = 0; i < (ssize_t)choices.nmemb && i < lines; ++i)
```

Comment:
In this case there is no off-by-one error since starting the program will
place the cursor on the next line after the prompt (i.e. the query line) and
therefore `min(choices.nmemb, lines)` lines needs to be emitted.

On Mon, Jan 04, 2016 at 01:40:33AM -0800, Calle Erlandsson wrote:
> > @@ -602,13 +553,9 @@ init_tty(void)
> >  		tty_putp(enter_ca_mode);
> >  
> >  	/* Emit enough lines to fit all choices. */
> > -	SLIST_FOREACH(choice, choices, choices) {
> > -		choices_count++;
> > +	for (i = 0; i < (ssize_t)choices.nmemb && i < lines; ++i)
> 
> Given https://github.com/thoughtbot/pick/pull/82/files#r48715231 is correct, should it be `lines - 1` here too?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/thoughtbot/pick/pull/82/files#r48715292

-- 
:wq